### PR TITLE
Fix/store session and visual bug + reaction final features

### LINKED ISSUE #125: Błąd z wartością początkową StoreContext
Po wstawieniu pod questions tablicy mockowej (plik StoreService.tsx) i po przejściu do zakładki zawierającej pytania (np. Pytania) nie ma żadnych pytań.

### LINKED ISSUE #211: Naprawić bugi tworzenia sesji
przy pierwszym odpaleniu po zbuildowaniu jest karta uczestnicy zamiast rozpocznij sesję
image
pierwsze tworzenie sesji nie redirectuje do uczestników.
odświeżanie strony czasami przekierowuje do create session nawet jeżeli sesja istnieje

### LINKED ISSUE #213: Redirect przy switchowaniu zakładek rozpocznij sesję/uczestnicy
When i refresh website being on UCZESTNICY I am redirected to another random page

### LINKED ISSUE #252: Zerowanie reakcji - progressbar
dodać progressbar mijającego czasu nad reakcjami i żeby było widać dla czego i kiedy reakcje się zerują

### LINKED ISSUE #321: W menu nie ma podkreślenia
Wystarczy dodać value do notifiable tab

### LINKED ISSUE #328: Sprawdzać czy kod zaproszenia z linku jest liczbą a nie tylko czy ma 7 znaków i odrzucić go jeśli nie jest liczbą
przykładowy błędny link
https://oi-project.herokuapp.com/student/code/aaaaaaa

### LINKED ISSUE #330: Po zaznaczeniu i odznaczeniu reakcji u wykładowcy nie czyści się po 15s
gdy zaznaczamy daną reakcje i ją potem odznaczymy po wyczyszczeniu reakcji to się nie wyczyści (a powinna)
jeżeli po odznaczeniu nie przyjdzie jakaś reakcja to nie włączy się timer 15s, więc reakcja po odznaczeniu może czekać w nieskończoność

### LINKED ISSUE #331: Usunięcie testowego guzika z wyświetlania reakcji

### PR DESCRIPTION
**Pull request by Derstilon**
image
I've added a progress bar over the reactions buttons.
Fixed an issue that reactions didn't clear after changing the mode.
Added store cleanup operation.
Fixed routing bug.
Added back topbar underline and enhanced the code for future routing.
Enhanced mockup code for better development experience.

### CODE REVIEW

**Review by ostatni5 — CHANGES_REQUESTED** [review]
A session is created but the view does not change.

obraz
obraz
obraz

**Review by michal367 — CHANGES_REQUESTED** [review]
Lecturer homepage doesn't work on HTTP deno server (it works but after clicking in menu I can go to homepage) - it is for sure problem with front routing
When I start session I am not redirected to session page (it only appears at first visit at website - maybe localstorage problem?)

When I refresh website I am still redirected to random page (not as many times as before but still it happens). Take a look at url. It still wanders between sites.

Reaction timer is starting even when students use only marked by lecturer reactions
After refreshing website marked reactions dissapear (is it intended?)
I think reactions progressbar should be more smooth

I think there should be no starting mock in questions

File: src/lecturer/components/sessionDashboardView/SessionDashboardView.tsx [hunk 1]
```diff
@@ -139,47 +139,19 @@ export function SessionDashboardView() {
...
-    //     socketEmiter.once("student_list", handleGetStudentList);
-    //     console.log("refresh whole list");
-    //     const payload: Payload = {
-    //         event: "get_student_list"
-    //     };
-    //     sendJsonMessage(payload); 
-    //     // return () => {
-    //     //     socketEmiter.removeListener("student_list", handleGetStudentList);
-    //     // };
-    // }, [handleGetStudentList, sendJsonMessage, socketEmiter]);
```
Comment (michal367) [inline src/lecturer/components/sessionDashboardView/SessionDashboardView.tsx]: Why did you remove it? @rozchlastywacz told me it should stay for later.

**Review by michal367 — APPROVED** [review]
Great job

obraz

File: server/server.ts [hunk 2]
```diff
@@ -29,7 +29,8 @@ app.use(async (ctx) => {
     catch (error) {
         if (error instanceof HttpError) {
             // console.log(ctx.request.url.pathname + " - " + error.message);
-            if (ctx.request.url.pathname.startsWith("/student/"))
+            if(ctx.request.url.pathname.startsWith("/favicon.ico")){
+            }else if (ctx.request.url.pathname.startsWith("/student/"))
```
Comment (ostatni5) [inline server/server.ts]: Empty if body?

File: src/common/util/routeToTabsValue.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "/lecturer/quiz":1,
+    "/lecturer/quizzes":1,
+    "/lecturer/question":2,
+    "/lecturer/questions":2,
+    "/lecturer/timestamp":3,
+    "/lecturer/stats":4,
+};
+export default function routeToTabsValue(route:string){
+    return routeValueRecord[route];
+}
```
Comment (ostatni5) [inline src/common/util/routeToTabsValue.ts]: This file is only useful for the lecturer move to lecturer/util or event better to folder connected with routing

File: src/student/components/joinSessionView/JoinSessionView.tsx [hunk 4]
```diff
@@ -22,6 +22,16 @@ export function JoinSessionView() {
     const [dialogOpen, setDialogOpen] = useState(false);
     const [dialogText, setDialogText] = useState(text.sessionEnded);
 
+    const getSessionID = () => {
+        let sessionId = "";
```
Comment (ostatni5) [inline src/student/components/joinSessionView/JoinSessionView.tsx]: Why not use a consistent style when naming variables like camelCase?

**Review by ostatni5 — COMMENTED** [review]
It works ok.

File: src/common/util/routeToTabsValue.ts (hunk 3 above)
Comment (Derstilon) [reply src/common/util/routeToTabsValue.ts]: image
new path

File: src/student/components/joinSessionView/JoinSessionView.tsx (hunk 4 above)
Comment (Derstilon) [reply src/student/components/joinSessionView/JoinSessionView.tsx]: To be fair this isn't id for the session, but an invitation code. I've changed it accordingly.

Comment (ostatni5) [reply src/student/components/joinSessionView/JoinSessionView.tsx]: XD

**Review by ostatni5 — APPROVED** [review]
PR with too many changes, it is hard to analyze code.

**Pull request merged**